Components/TransactionReport/TransactionReport.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * (c) shopware AG <info@shopware.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SwagPaymentPayPalUnified\Components\TransactionReport;
+
+use DateTime;
+use Doctrine\DBAL\Connection;
+use GuzzleHttp\Client;
+use GuzzleHttp\RequestOptions;
+use PDO;
+use Shopware\Models\Order\Status;
+
+final class TransactionReport
+{
+    const TRANSACTION_REPORT_TABLE = 'swag_payment_paypal_unified_transaction_report';
+    const API_IDENTIFIER = '9b6f559b-5ca1-4969-b23d-e0aa2c01d562';
+    const POST_URL = 'https://api.shopware.com';
+    const POST_URL_ENDPOINT = '/shopwarepartners/reports/technology';
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * @param int $orderId
+     *
+     * @return void
+     */
+    public function reportOrder($orderId)
+    {
+        $this->connection->createQueryBuilder()
+            ->insert(self::TRANSACTION_REPORT_TABLE)
+            ->setValue('order_id', ':orderId')
+            ->setParameter('orderId', $orderId)
+            ->execute();
+    }
+
+    /**
+     * @param string $shopwareVersion
+     *
+     * @return void
+     */
+    public function report($shopwareVersion, Client $client)
+    {
+        $reportResult = $this->getReportResult($this->getReportedOrderIds());
```

Comment:
why not creating the client in this method? I guess you do not want to add those method parameters to the constructor, as they are not used by every public method?

Reply:
for testin